src/components/DefineDissolution/DestroyCertificate.vue

```diff
@@ -0,0 +1,91 @@
+<template>
+  <div>
+    <div v-if="isSummary" class="destroy-certificate-summary">
+      <div v-if="getHasCertificateDestroyed" class="d-flex">
+        <span>
+          <v-icon class="destroy-certificate-success-chk ml-n1 pr-2" color="successCheckmark">mdi-check</v-icon>
+        </span>
+        <span v-html="destroyCertificateDescription" />
+      </div>
+      <div v-else>(Not entered)</div>
+    </div>
+    <v-card v-else flat class="rounded-4">
+      <div class="section-container pt-11" :class="{ 'invalid-section': showErrorSummary }">
+        <v-row no-gutters>
+          <v-col md="3">
+            <label class="destroy-certificate-title">Delete and/or Destroy Certificates</label>
+          </v-col>
+          <v-col md="9">
+            <v-checkbox
+              class="destroy-certificate-checkbox mt-0"
+              v-model="hasCertificateDestroyed"
+              @change="setHasCertificateDestroyed"
+            >
+              <template slot="label">
+                <span class="destroy-certificate-option"
+                  v-html="destroyCertificateDescription"
+                  :class="{'error-text': showErrorSummary}" />
+              </template>
+            </v-checkbox>
+          </v-col>
+        </v-row>
+      </div>
+    </v-card>
+  </div>
+</template>
+
+<script lang="ts">
+// Libraries
+import { Component, Vue, Prop } from 'vue-property-decorator'
+import { Action, Getter } from 'vuex-class'
+
+// Interfaces
+import { ActionBindingIF } from '@/interfaces'
+
+@Component
+export default class DestroyCertificate extends Vue {
+  @Prop({ default: false })
+  private readonly showErrorSummary: boolean
+
+  @Prop({ default: false })
+  private readonly isSummary: boolean
+
+  private hasCertificateDestroyed: boolean = false
+
+  private destroyCertificateDescription: string = `Certificates of incorporation, name change, and amalgamation for
```

Comment:
Could probably be `readonly`.

Reply:
Globally, we need to think about what's private or public, readonly or not, and whether we need to explicitly declare strings as strings.

My thoughts:
- things that should be only-internal to the class should be private
- variables that shouldn't be reassigned should be `const`; properties that shouldn't be reassigned should be `readonly`; it's a subtle different, see https://www.google.com/search?q=typescript+const+vs+readonly
- sometimes, the compiler will complain when you do `name: string = 'SB'` because it's redundant, so I generally drop the type when it's evident from the assigned value

In this case, the string looks more like a property than a variable to me, so `readonly`  is probably more appropriate. But it doesn't need to be protected inside this class so private isn't needed.

(Note that, at this time, visibility may or may not be checked at compile time or run time.)

Reply:
This is a really interesting topic for sure, would love to learn more on it. Perhaps a good discussion for a tech demo or something!

Reply:
Haha, I wish I knew all the answers! This one would take some reading to present well :)

Reply:
cannot make it as const since its on class scope. vs code gives this error `A class member cannot have the 'const' keyword`. So changing it as readonly